diff --git a/docker/Dockerfile b/docker/Dockerfile
index 4cd4def..95cca6d 100644
--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -2,6 +2,12 @@ ARG BASE_IMAGE=docker.io/ubuntu:22.04
 FROM $BASE_IMAGE
 
 ARG BASE_IMAGE
+ARG RELEASE_PACKAGE
+ARG RELEASE_VERSION
+ARG RELEASE_ARCH
+ARG RELEASE_REPO
+ARG RELEASE_PACK_URL
+ARG OS_NAME
 LABEL org.opencontainers.image.authors="GitLab Distribution Team <distribution-be@gitlab.com>" \
       org.opencontainers.image.documentation="https://docs.gitlab.com/ee/install/docker/" \
       org.opencontainers.image.source="https://gitlab.com/gitlab-org/omnibus-gitlab" \
diff --git b/docker/RELEASE b/docker/RELEASE
new file mode 100644
index 0000000..8b05830
--- /dev/null
+++ b/docker/RELEASE
@@ -0,0 +1,7 @@
+RELEASE_PACKAGE=${RELEASE_PACKAGE}
+RELEASE_VERSION=${RELEASE_VERSION}
+RELEASE_ARCH=${RELEASE_ARCH}
+RELEASE_REPO=${RELEASE_REPO}
+RELEASE_PACK_URL=${RELEASE_PACK_URL}
+OS_NAME=${OS_NAME}
+DOWNLOAD_URL_FULL=https://packages.gitlab.com/gitlab/${RELEASE_REPO}/packages/${RELEASE_PACK_URL}/${OS_NAME}/${RELEASE_PACKAGE}_${RELEASE_VERSION}_${RELEASE_ARCH}.deb/download.deb
\ No newline at end of file
diff --git a/docker/assets/setup b/docker/assets/setup
index 03e8c6d..0827fd0 100755
--- a/docker/assets/setup
+++ b/docker/assets/setup
@@ -18,7 +18,7 @@ else
   echo "Unknown TARGETARCH: DOWNLOAD_URL not set"
 fi
 
-DOWNLOAD_URL=${DOWNLOAD_URL} CI_JOB_TOKEN=${CI_JOB_TOKEN} /assets/download-package && dpkg -i /tmp/gitlab.deb 
+DOWNLOAD_URL=${DOWNLOAD_URL_FULL} CI_JOB_TOKEN=${CI_JOB_TOKEN} /assets/download-package && dpkg -i /tmp/gitlab.deb 
 rm -rf /tmp/gitlab.deb /var/lib/apt/lists/*
 
 unset DOWNLOAD_URL_amd64
